db/db_impl.cc

```diff
@@ -4034,18 +4038,30 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
   {
     if (creating_new_log) {
       EnvOptions opt_env_opt =
-          env_->OptimizeForLogWrite(env_options_, db_options_);
-      s = env_->NewWritableFile(
+	env_->OptimizeForLogWrite(env_options_, db_options_);
+      if (recycle_log_number) {
+	Log(InfoLogLevel::INFO_LEVEL, db_options_.info_log,
+	    "reusing log %" PRIu64 " from recycle list\n",
+	    recycle_log_number);
+	s = env_->ReuseWritableFile(
+          LogFileName(db_options_.wal_dir, new_log_number),
+	  LogFileName(db_options_.wal_dir, recycle_log_number),
+	  &lfile,
+	  opt_env_opt);
+      } else {
+	s = env_->NewWritableFile(
           LogFileName(db_options_.wal_dir, new_log_number), &lfile,
-          opt_env_opt);
+	  opt_env_opt);
+      }
       if (s.ok()) {
-        // Our final size should be less than write_buffer_size
-        // (compression, etc) but err on the side of caution.
-        lfile->SetPreallocationBlockSize(
+	 // Our final size should be less than write_buffer_size
+	 // (compression, etc) but err on the side of caution.
+	 lfile->SetPreallocationBlockSize(
             1.1 * mutable_cf_options.write_buffer_size);
         unique_ptr<WritableFileWriter> file_writer(
             new WritableFileWriter(std::move(lfile), opt_env_opt));
-        new_log = new log::Writer(std::move(file_writer));
+	new_log = new log::Writer(&db_options_, std::move(file_writer),
```

Comment:
Looks like you changed the indenting. If you run "make format" in command line, it will fix most of the indenting problems like this.